File: Examples/CSharp/SimpleTransformix.cs
using System;
using itk.simple;

namespace itk.simple.examples {
    class SimpleTransformix {
        static void Main(string[] args) {
            try {
                if (args.Length < 4) {
                    Console.WriteLine("Usage: SimpleTransformix <fixedImage> <movingImage> <inputImage> <parameterFile> <output>");
                    return;
                }
                // Make transform
                ElastixImageFilter silx;
                ImageFileReader reader = new ImageFileReader();
                reader.SetFileName(args[0]);
                silx.SetFixedImage(reader.Execute());
                reader.SetFileName(args[1]);
                silx.SetMovingImage(reader.Execute());
                silx.SetParameterMap(ReadParameterFile(args[3]));
                silx.Execute();

                // Instantiate transformix
                TransformixImageFilter stfx;

                // Read input
                reader.SetFileName(args[4]);
                stfx.SetInputImage(reader.execute());
                stfx.SetTransformParameterMap(silx.GetTransformParameterMap());

                // Perform warp
                stfx.LogToConsoleOn();
                stfx.Execute();

                // Write result image
                sitk.WriteImage(stfx.GetResultImage(), args[5]);
                

            } catch (Exception e) {
                Console.WriteLine(e);
            }
        }
    }
}

File: Examples/Python/SimpleTransformix.py
import SimpleITK as sitk
import sys

# Make transform 
elastixImageFilter = sitk.ElastixImageFilter()
elastixImageFilter.SetFixedImage(sitk.ReadImage(str(sys.argv[1])))
elastixImageFilter.SetMovingImage(sitk.ReadImage(str(sys.argv[2])))
elastixImageFilter.SetParameterMap(sitk.ReadParameterFile(str(sys.argv[3])))
elastixImageFilter.LogToConsoleOn()
elastixImageFilter.Execute()

# Instantiate SimpleTransformix
transformixImageFilter = sitk.TransformixImageFilter()

# Read Input
transformixImageFilter.SetInputImage(sitk.ReadImage(str(sys.argv[4])))
transformixImageFilter.SetParameterMap(elastixImageFilter.GetTransformParameterMap())

# Perform warp
transformixImageFilter.LogToConsoleOn()
transformixImageFilter.Execute()

# Write result image
sitk.WriteImage(transformixImageFilter.GetResultImage(), str(sys.argv[5]))
